balboa-agent/src/main/scala/com/socrata/balboa/agent/HttpMetricQueue.scala

```diff
@@ -0,0 +1,87 @@
+import com.stackmob.newman.ApacheHttpClient
+import com.stackmob.newman.dsl.POST
+import com.stackmob.newman.response.HttpResponseCode.Ok
+import com.typesafe.scalalogging.slf4j.StrictLogging
+import org.json4s.jackson.JsonMethods.{pretty, render}
+import org.json4s.{DefaultFormats, Extraction, Formats}
+
+import scala.concurrent.duration._
+import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+case class HttpMetricQueue(balboaHttpURL: String, timeout: Duration, maxRetryWait: Duration)
+  extends MetricQueue with StrictLogging {
+
+  val StartingWaitDuration = 50.millis
+  val Utf8 = "UTF_8"
+
+  implicit val httpClient = new ApacheHttpClient
+  implicit val jsonFormats: Formats = DefaultFormats
+  implicit val executionContext = ExecutionContext.global
+
+  logger info s"Initializing HttpMetricQueue targeting $balboaHttpURL"
+
+  /**
+   * Accepts a metric to transmit to the remote metric store.
+   *
+   * @param entity Entity which this Metric belongs to (ex: a domain).
+   * @param name Name of the Metric to store.
+   * @param value Numeric value of this metric.
+   * @param timestamp Time when this metric was created.
+   * @param recordType Type of metric to add, See [[Metric.RecordType]] for more information.
+   */
+  override def create(entity: IdParts,
+             name: IdParts,
+             value: Long,
+             timestamp: Long = new Date().getTime,
+             recordType: Metric.RecordType = Metric.RecordType.AGGREGATE): Unit = {
+
+    val metricToWrite = EntityJSON(timestamp, Map(name.toString -> MetricJSON(value, recordType.toString)))
+    val url = new URL(s"$balboaHttpURL/metrics/${URLEncoder.encode(entity.toString, Utf8)}")
+    val request = POST(url).setBody(pretty(render(Extraction.decompose(metricToWrite))))
+
+    var waitInLoop = StartingWaitDuration
+
+    // Originally the transport mechanism for publishing metrics from
+    // balboa-agent to the metric store was ActiveMQ. ActiveMQ has a built in
+    // infinite retry for publishing to the queue. During the transition from
+    // ActiveMQ to sending metrics via HTTP, a strategy of replicating
+    // ActiveMQ's behavior of getting stuck here and infinitely retrying was
+    // taken to make the code transition simpler.
+    //
+    // An improvement on this naive infinite retry policy could go 1 of two ways:
+    // 1. Adapt `balboa-http` to accept all transmissions and persist the ones
+    //    it can't parse, retrying them later (when it's been upgraded for
+    //    example). This would safer for keeping accurate metrics.
+    // 2. Make `balboa-agent` respond differently to different HTTP error
+    //    codes. Possibly, for example (but not exhaustively) 404, 500s are
+    //    infinite retry, but 400s are written to a file to be retried in the
+    //    future.
+    while (true) {
+      val requestWithTimeout = Future { Await.result(request.apply, timeout) }
```

Comment:
This seems odd to me. The purpose of `Await` is to serialize a `Future`. Which means `request.apply` is returning a `Future`. But you are wrapping `Await` back into another `Future`.

Since `requestWithTimeout` is a `Future`, that means it is executing in the background, and the loop will wait, come back here and start another `request.apply` in the background. So if the server is slow, there is a race condition where multiple requests can be submitted.